core/types/receipt.go

```diff
@@ -499,7 +531,7 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 			// Deriving the signer is expensive, only do if it's actually needed
 			from, _ := Sender(signer, txs[i])
 			nonce := txs[i].Nonce()
-			if rs[i].DepositNonce != nil {
+			if txs[i].IsDepositTx() && rs[i].DepositNonce != nil {
```

Comment:
`txs[i].IsDepositTx()` seems to be not necessary.

Reply:
We changes the way to store `DepositNonce`, so the `DepositNonce` will not be nil and the default value will be 0 for non-depositTxs, so add `txs[i].IsDepositTx()` to determine the tx type.